test/extended/util/prometheus/helpers.go

```diff
@@ -286,18 +319,26 @@ func RunQueries(ctx context.Context, prometheusClient prometheusv1.API, promQuer
 // ExpectURLStatusCodeExec attempts connection to url returning an error
 // upon failure or if status return code is not equal to any of the statusCodes.
 func ExpectURLStatusCodeExec(url string, statusCodes ...int) error {
-	cmd := fmt.Sprintf("curl -k -s -o /dev/null -w '%%{http_code}' %q", url)
-	output, err := exec.Command("bash", "-e", "-c", cmd).Output()
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   time.Duration(10 * time.Second),
+	}
+
+	resp, err := client.Get(url)
 	if err != nil {
-		return fmt.Errorf("host command failed: %v\n%s", err, output)
+		return err
 	}
+	defer resp.Body.Close()
+
 	for _, statusCode := range statusCodes {
-		if string(output) == strconv.Itoa(statusCode) {
+		if resp.StatusCode == statusCode {
 			return nil
 		}
 	}
-
-	return fmt.Errorf("last response from server was not in %v: %s", statusCodes, output)
+	return fmt.Errorf("last response from server was not in %v: %d", statusCodes, resp.StatusCode)
```

Comment:
(nit) while we are at it
```suggestion
	return fmt.Errorf("%s: last response from server was not in %v: %d", url, statusCodes, resp.StatusCode)
```